### ISSUE TITLE
Can license-checker be excluded by spring boot fat jar without breaking anything?

### ISSUE THREAD

**Issue by knoobie** [vaadin/license-checker#91]
Title says it all. Can the license checker (a 4mb jar) be removed from production builds? Is it needed at runtime?

Additionally the license-checker jar contains jna, jna-platform and oshi-core. Those are already present within a spring boot fat jar and are probably the reason why this jar is so big.

Example based on the latest 23.1.0.beta1 with license checker 1.4.1

![image](https://user-images.githubusercontent.com/3968629/166936101-5c24478f-06cf-4c2e-bdf8-eb3917589798.png)

**Comment by Artur-** [comment]
Yes, it is needed for the production build but not in the built jar. I wonder if we could automatically exclude it in some sensible way

**Comment by knoobie** [comment]
Could it be added as dependency to the maven/gradle plugin?

**Comment by Artur-** [comment]
If you look at a  23.0.0 application you have
```
   19562  Stored    19562   0% 02-04-2022 14:36 50a588d4  BOOT-INF/lib/license-checker-1.2.2.jar
 1343495  Stored  1343495   0% 11-08-2021 20:28 0b31df72  BOOT-INF/lib/jna-platform-5.10.0.jar
 1756400  Stored  1756400   0% 11-08-2021 20:28 ed29c485  BOOT-INF/lib/jna-5.10.0.jar
  890716  Stored   890716   0% 01-20-2022 08:59 edadea74  BOOT-INF/lib/oshi-core-6.1.0.jar
```

Similarly for 23.1.0.beta1 you have
```
 3799999  Stored  3799999   0% 05-03-2022 13:56 4a08b143  BOOT-INF/lib/license-checker-1.4.1.jar
 1352301  Stored  1352301   0% 03-27-2022 11:44 63c7a3d8  BOOT-INF/lib/jna-platform-5.11.0.jar
 1825174  Stored  1825174   0% 03-27-2022 11:44 d2f9fa0e  BOOT-INF/lib/jna-5.11.0.jar
  906283  Stored   906283   0% 04-10-2022 19:43 a1026332  BOOT-INF/lib/oshi-core-6.1.6.jar
```

So we can see that the license checker has always been included but version 1.4.1 is much larger than 1.2.2. It should not be though and it should not be impossible to fix that. It currently bundles the jna and oshi jars inside itself to get around some osgi issue. That should be solved in another way.

Now the real difference in size between 23.0 and 23.1 is that 23.1 includes spreadsheet which brings in poi
```
 2920513  Stored  2920513   0% 03-12-2022 14:21 f1d529fa  BOOT-INF/lib/poi-5.2.2.jar
 1897101  Stored  1897101   0% 03-12-2022 14:21 1d4e3514  BOOT-INF/lib/poi-scratchpad-5.2.2.jar
 1977285  Stored  1977285   0% 03-12-2022 14:24 5aa106b8  BOOT-INF/lib/poi-ooxml-5.2.2.jar
13060187  Stored 13060187   0% 03-12-2022 14:22 c0ffff27  BOOT-INF/lib/poi-ooxml-full-5.2.2.jar
```

**Comment by Artur-** [comment]
I would imagine the size does not matter that much for development and maybe the proper solution here would be to exclude development dependencies from the production jar/war. There is also vaadin-dev-server nobody needs in production. Also it should be possible to exclude components that are not used instead of apways including everything ”just in case”. Then you would not get poi unless you are actually using spreadsheet

**Comment by knoobie** [comment]
My concern comes from the transitive dependencies the license checker has and aren't really needed for free components because in my government project every dependency has to be known, checked by our security department and passing our architecture board. (Trust me; it's a hassle)

Having three new 1-2mb big dependencies with a lot of possible functions is going to create questions 😅

To summarize, there are currently two problems: 
- the license checker is big (minor)
- the license checker and it's transitive dependencies are added even tho there aren't any components used or added that need it (major in my case)

**Comment by knoobie** [comment]
The following would exclude it from the production artifact (but I'm not sure if this would break anything; especially if the feature flag for offline check is activated)

```xml

    <dependency>
      <groupId>com.vaadin</groupId>
      <artifactId>vaadin-core</artifactId>
      <exclusions>
        <exclusion>
          <groupId>com.vaadin</groupId>
          <artifactId>license-checker</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>com.vaadin</groupId>
      <artifactId>vaadin-spring-boot-starter</artifactId>
      <exclusions>
        <exclusion>
          <groupId>com.vaadin</groupId>
          <artifactId>vaadin-core</artifactId>
        </exclusion>
      </exclusions>
    </dependency>

```
